src/storage/log/aof/log.rs

```diff
@@ -298,19 +297,19 @@ impl Aol {
     }
 
     pub fn close(&mut self) -> Result<()> {
-        let _lock = self.mutex.write();
+        let _lock = self.mutex.lock();
         self.active_segment.close()?;
         Ok(())
     }
 
     // Returns the current offset within the segment.
     pub fn offset(&self) -> Result<u64> {
-        let _lock = self.mutex.read();
+        let _lock = self.mutex.lock();
         Ok((self.active_segment_id * self.opts.max_file_size) + self.active_segment.offset())
```

Comment:
```suggestion
        Ok(self.calculate_offset() + self.active_segment.offset())
```

Probably calculate_offset is handy here too?

Reply:
Thanks @mumoshu !